File: pedidos/src/main/java/br/com/bota/event/ProdutoEntregaEvent.java
package br.com.bota.event;

import br.com.bota.dto.EntregaDTO;
import io.micronaut.configuration.kafka.annotation.KafkaClient;
import io.micronaut.configuration.kafka.annotation.KafkaKey;
import io.micronaut.configuration.kafka.annotation.Topic;

@KafkaClient
public interface ProdutoEntregaEvent {

    @Topic("pedido_entrega")
    void enviarPedidoAoEntregador(@KafkaKey long key, EntregaDTO entregaDTO);
}

File: loja-kotlin/kpedidos/src/main/kotlin/br/com/bota/event/client/ProdutoEntregaEvent.kt
package br.com.bota.event.client

import br.com.bota.lojalib.dto.EntregaDTO
import io.micronaut.configuration.kafka.annotation.KafkaClient
import io.micronaut.configuration.kafka.annotation.KafkaKey
import io.micronaut.configuration.kafka.annotation.Topic

@KafkaClient
interface ProdutoEntregaEvent {
    @Topic("pedido_entrega")
    fun enviarPedidoAoEntregador(@KafkaKey key : Long, entregaDTO: EntregaDTO)
}